src/firefly/java/edu/caltech/ipac/firefly/util/ImageSetConverter.java

```diff
@@ -0,0 +1,209 @@
+        }
+        imageSetToJSONArray = createImageJSONArray(masterTable);
+        //test the result
+     /*   try {
+            writeJSONArrayToFile(imageSetToJSONArray,  "/Users/zhang/IRSA_Dev/testingData/irsa-master-table.json");
+        }
+        catch(Exception e){
+            e.printStackTrace();
+        }*/
+    }
+
+    /**
+     * This method reads in the master-table and then create a JSONArray based on the table data.
+     * @return
+     * @throws IOException
+     * @throws FailedRequestException
+     */
+    private JSONArray createImageJSONArray(String masterTable) throws IOException, FailedRequestException {
+        DataGroup inDg = getDataFromMasterTable(masterTable);
+        DataType[] dataTypes = inDg.getDataDefinitions();
+        List<DataObject> dataRows = inDg.values();
+
+
+        for (int i=0; i<dataRows.size(); i++){
+            imageSetToJSONArray.add(dataObjectToJSONObject(dataRows.get(i),dataTypes));
+        }
+
+        LOG.info("JSONArray is created successfully");
+
+
+        return imageSetToJSONArray;
+    }
+
+    /**
+     * This method finds the plotRequestParams.  It uses predefined map to map the data in the master table.
+     * For example, the apiType in master table means Service for the plotRequest parameter.
+     * @param row
+     * @param dataTypes
+     * @return
+     */
+    private HashMap<String, String> getMappedPlotRequestParam(DataObject row, DataType[] dataTypes){
+
+        HashMap<String, String> params = new HashMap<>();
+
+        String[] plotParamKeys =  paramMaps.keySet().toArray(new String[0]);
+        for (int i=0; i<plotParamKeys.length; i++) {
+            String key= plotParamKeys[i];
+            for (int j = 0; j < row.size(); j++) {
+                String colName = dataTypes[j].getKeyName();
+                if(colName.equalsIgnoreCase(key)){
+                    params.put( paramMaps.get(key), row.getFormatedData(dataTypes[j]));
+                    break;
+                }
+            }
+
+        }
+        return params;
+    }
+
+    /**
+     * This method process each row data in the master-table.  Each row makes one JSONObject. All rows make
+     * an JSONArray of the size = number of rows
+     * @param row
+     * @param dataTypes
+     * @return
+     */
+    private JSONObject dataObjectToJSONObject(DataObject row,  DataType[] dataTypes){
+        JSONObject jsonObject = new JSONObject();
+
+
+        for (int i=0; i<dataTypes.length; i++){
+            Class cls = dataTypes[i].getDataType();
+            Object obj = row.getDataElement(dataTypes[i]);
+            if (cls==null){
+              jsonObject.put(dataTypes[i].getKeyName(),  obj);
+              continue;
+            }
+
+            switch (cls.getSimpleName().toLowerCase()){
+                case "string":
+                    String val = obj==null? null:((String) obj).trim();
+
+                    if (dataTypes[i].getKeyName().equalsIgnoreCase("missionLabel (project)")){
```

Comment:
I think we need to talk about this. The mapping and having the hard-coded values is fine for now but it really should come from metadata API (future) or at least from properties so we don't have to change this code to make it work. Let's discuss this together.